gnucash/import-export/aqb/dialog-ab-select-imexporter.c

```diff
@@ -0,0 +1,257 @@
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org
+ */
+
+/**
+ * @internal
+ * @file dialog-ab-select-imexporter.h
+ * @brief Dialog to select AQBanking importer/exporter and format profile.
+ * @author  Copyright (C) 2022 John Ralls <jralls@ceridwen.us>
+ */
+
+#include <config.h>
+
+#include <glib/gi18n.h>
+#include "dialog-ab-select-imexporter.h"
+#include <dialog-utils.h>
+
+static QofLogModule log_module = G_LOG_DOMAIN;
+
+struct _GncABSelectImExDlg
+{
+    GtkWidget *dialog;
+    GtkWidget *parent;
+    GtkListStore *imexporter_list;
+    GtkListStore *profile_list;
+    GtkWidget *select_imexporter;
+    GtkWidget *select_profile;
+    GtkWidget *ok_button;
+
+    AB_BANKING* abi;
+};
+
+// Expose the selection handlers to GtkBuilder.
+G_MODULE_EXPORT gboolean imexporter_changed(GtkTreeSelection* sel,
+                                            gpointer data);
+G_MODULE_EXPORT gboolean profile_changed(GtkTreeSelection* sel, gpointer data);
+
+static gboolean
+clear_model_row (GtkTreeModel* model, GtkTreePath* path,
+                 GtkTreeIter* iter, gpointer data)
+{
+    char *name, *desc;
+    g_return_val_if_fail (model && iter, TRUE);
+    gtk_tree_model_get (model, iter, 0, &name, 1, &desc, -1);
+    g_free (name);
+    g_free (desc);
+    return FALSE;
+}
+
+static void
+clear_tree_model (GtkTreeModel* model)
+{
+    gtk_tree_model_foreach (model, clear_model_row, NULL);
+    gtk_list_store_clear (GTK_LIST_STORE (model));
+}
+
+static void
+populate_tree_model (GtkTreeModel* model, GList* entries)
+{
+    clear_tree_model (model);
+    for (GList* node = entries; node; node = g_list_next (node))
+    {
+        AB_Node_Pair *pair = (AB_Node_Pair*)(node->data);
+        GtkTreeIter iter;
+        gtk_list_store_insert_with_values (GTK_LIST_STORE (model),
+                                           &iter, -1,
+                                           0, pair->name,
+                                           1, pair->descr,
+                                           -1);
+        g_slice_free1 (sizeof(AB_Node_Pair), pair);
+    }
+}
+
+GncABSelectImExDlg*
+gnc_ab_select_imex_dlg_new (GtkWidget* parent, AB_BANKING* abi)
+{
+    GncABSelectImExDlg* imexd;
+    GtkBuilder* builder;
+    GList* imexporters;
+    GtkTreeSelection *imex_select = NULL, *prof_select = NULL;
+
+    g_return_val_if_fail (abi, NULL);
+    imexporters = gnc_ab_imexporter_list (abi);
+    g_return_val_if_fail (imexporters, NULL);
+    imexd = g_new0(GncABSelectImExDlg, 1);
+    imexd->parent = parent;
+    imexd->abi = abi;
+
+    g_signal_connect (parent, "destroy",
+                      G_CALLBACK (gtk_widget_destroyed), &imexd->parent);
+    builder = gtk_builder_new();
+    gnc_builder_add_from_file (builder, "dialog-ab.glade", "imexporter-list");
+    gnc_builder_add_from_file (builder, "dialog-ab.glade", "profile-list");
+    gnc_builder_add_from_file (builder, "dialog-ab.glade",
+                               "aqbanking-select-imexporter-dialog");
```

Comment:
Not sure it matters but all the other dialogs in glade file use '_' instead of '-' for name.

Reply:
They're just strings to the compiler and GtkBuilder doesn't seem to care. Does Glade care? (I don't use it.)

Reply:
I'll leave this in for now, it's easily changed if Glade gives you trouble.